File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k = k % nums.size(); 
        vector<int> inter; 
        
        for(int i = nums.size() - k; i < nums.size(); i++){
            inter.push_back(nums[i]); 
        }
        for(int i = 0; i < nums.size() - k; i++){
            inter.push_back(nums[i]); 
        }
        for(int i = 0; i < nums.size(); i++){
            nums[i] = inter[i]; 
        }

        
        

        
    }
};

File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        #first rotate the array 
        nums.reverse()
        k = k % len(nums)
        left, right = k,len(nums)-1
        print(nums)
        while left < right: 
            nums[right], nums[left] = nums[left], nums[right]
            right-=1
            left+= 1

        first, second = 0, k-1
        while first < second: 
            nums[first], nums[second] = nums[second], nums[first]
            first += 1
            second -= 1

          
        

